web/packages/teleterm/src/ui/Search/useSearch.test.ts

```diff
@@ -63,4 +63,35 @@ describe('sortResults', () => {
     expect(fooMatches[0].score).toBeGreaterThan(quuxMatch.score);
     expect(fooMatches[1].score).toBeGreaterThan(quuxMatch.score);
   });
+
+  it('limits the results', () => {
+    const servers = Array(10)
+      .fill(undefined)
+      .map(() =>
+        makeResourceResult({
+          kind: 'server',
+          resource: makeServer({
+            labelsList: makeLabelsList({ foo: 'bar' }),
+          }),
+        })
+      );
+
+    // This item has a lower score
+    servers.unshift(
+      makeResourceResult({
+        kind: 'server',
+        resource: makeServer({
+          labelsList: makeLabelsList({ abc: 'bar123456' }),
+        }),
+      })
+    );
+
+    const sorted = sortAndLimitResults(servers, 'bar');
+
+    expect(sorted).toHaveLength(10);
+    // the item with the lowest score is not included
+    expect(
+      sorted.find(s => s.labelMatches.find(a => a.labelName === 'abc'))
```

Comment:
What would be also nice to test is adding the top ranked result as 12th result and check that it's returned as the first one.

The problem with the previous implementation was that results were limited before they were scored. If this happened again, this test wouldn't catch it as the last element would not be included in the final result no matter what.